src/Ztm.WebApi/AddressPools/LessUsageFirstChoser.cs

```diff
@@ -1,31 +1,29 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace Ztm.WebApi.AddressPools
 {
     public sealed class LessUsageFirstChoser : IAddressChoser
     {
-        public ReceivingAddress Choose(IEnumerable<ReceivingAddress> receivingAddress)
+        public ReceivingAddress Choose(IEnumerable<ReceivingAddress> addresses)
         {
-            var chosen = receivingAddress.First();
-            var currentUsage = chosen.ReceivingAddressReservations == null
-                ? 0
-                : chosen.ReceivingAddressReservations.Count;
-
-            foreach (var candidate in receivingAddress)
+            if (addresses == null)
             {
-                var usage = candidate.ReceivingAddressReservations == null
-                    ? 0
-                    : candidate.ReceivingAddressReservations.Count;
+                throw new ArgumentNullException(nameof(addresses));
+            }
 
-                if (usage < currentUsage)
-                {
-                    chosen = candidate;
-                    currentUsage = usage;
-                }
+            if (!addresses.Any())
+            {
+                throw new ArgumentOutOfRangeException(nameof(addresses), "Addresses could not be empty.");
```

Comment:
I think `ArgumentOutOfRangeException` is not quite correct here. We should use `ArgumentException` instead.